File: Libs/PersistenceLib/Domains/OrderApi/Product.cs
//-----------------------------------------------------------------------
// <copyright file="Product.cs" website="Patrikduch.com">
//     Copyright 2019 (c) Patrikduch.com
// </copyright>
// <author>Patrik Duch</author>
//-----------------------------------------------------------------------

using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;

namespace PersistenceLib.Domains.OrderApi
{
    /// <summary>
    /// Entity that represents products
    /// </summary>
    [Table("Product")]
    public class Product
    {
        /// <summary>
        /// Gets or sets product`s identifier
        /// </summary>
        public int Id { get; set; }
        /// <summary>
        /// Gets or sets product`s name
        /// </summary>
        public string Name { get; set; }
        /// <summary>
        /// Gets or sets product`s price
        /// </summary>
        public int Price { get; set; }
        /// <summary>
        /// Gets or sets product`s deletion state
        /// </summary>
        public bool IsDeleted { get; set; }

        public ICollection<OrderProduct> OrderProducts { get; set; }
    }
}

File: ReactApp/Frontend/view-models/Product.ts
//-------------------------------------------------------------------------------------------
// <copyright file="Product.ts" website="Patrikduch.com">
//     Copyright 2019 (c) Patrikduch.com
// </copyright>
// <author>Patrik Duch</author>
// View model for representation of products
//--------------------------------------------------------------------------------------------

export default class Product {
    private id: number;
    private name: string;
    private price: number;
    private isDeleted: boolean;

    constructor(id: number, name: string, price: number, isDeleted: boolean) {
        this.id = id;
        this.name = name;
        this.price = price;
        this.isDeleted = isDeleted;
    }

    get getId() {
        return this.id;
    }

    get getName() {
        return this.name;
    }

    get getPrice() {
        return this.price;
    }

    get getIsDeleted() {
        return this.isDeleted;
    }
}